cmd/miniooni/main.go

```diff
@@ -191,9 +176,19 @@ func main() {
 	if globalOptions.proxy != "" {
 		proxyURL = mustParseURL(globalOptions.proxy)
 	}
-	sess := session.New(
-		logger, softwareName, softwareVersion, workDir, proxyURL, tlsConfig,
-	)
+
+	sess, err := engine.NewSession(engine.SessionConfig{
+		AssetsDir:       tempDir,
+		Logger:          logger,
+		ProxyURL:        proxyURL,
+		SoftwareName:    softwareName,
+		SoftwareVersion: softwareVersion,
+		TempDir:         tempDir,
+		TLSConfig:       tlsConfig,
+	})
```

Comment:
This is one of the first reasons why I introduced this API: I want to make sure that when we construct a new session we are passing correct parameters and the mandatory ones are there. Also, this setting of options through a struct is more easy to get right than instead passing a long list of options to a constructor.